partiql-conformance-test-generator/src/schema.rs

```diff
@@ -0,0 +1,102 @@
+use crate::schema::TestCaseKind::Parse;
+use codegen::{Function, Module, Scope};
+
+/// Conformance test document containing namespaces and/or tests
+pub struct TestDocument {
+    pub(crate) namespaces: Namespaces,
+    pub(crate) test_cases: TestCases,
+}
+pub type Namespaces = Vec<Namespace>;
+
+/// Namespace can contain other namespaces and/or tests
+pub struct Namespace {
+    pub(crate) name: String,
+    pub(crate) namespaces: Namespaces,
+    pub(crate) test_cases: TestCases,
+}
+pub type TestCases = Vec<TestCase>;
+
+/// Test cases have a `test_name` and a PartiQL `statement` along with additional test fields stored
+/// depending on the `test_kind`.
+pub struct TestCase {
+    pub(crate) test_name: String,
+    pub(crate) statement: String,
+    pub(crate) test_kind: TestCaseKind,
+}
+
+/// Test case kind
+///
+/// Currently, just supports `Parse` test cases. In the future, other test case variants will be
+/// added (e.g. evaluation, type-checking)
+pub enum TestCaseKind {
+    Parse(ParseTestCase),
+}
+
+/// Test case to test the parsing behavior of a PartiQL statement
+pub struct ParseTestCase {
+    pub(crate) parse_assertions: ParseAssertions,
+}
+
+pub enum ParseAssertions {
+    ParsePass,
+    ParseFail,
+}
+
+impl TestDocument {
+    /// Converts a `TestDocument` into a `Scope`
+    pub fn generate_scope(&self) -> Scope {
```

Comment:
Long term, I think it would be better to have the generation not be an impl on the schema objects themselve, but maybe rather on a `Generator` object. This way if additional state is needed it can be present on the `Generator` and doesn't need to be threaded through each call specifically.

But this is fine for this PR; we can refactor as we learn more.